lib/algoliasearch-rails.rb

```diff
@@ -920,13 +923,16 @@ def algolia_find_in_batches(batch_size, &block)
       end
     end
 
-    def attribute_changed_method(attr)
-      if defined?(::ActiveRecord) && ActiveRecord::VERSION::MAJOR >= 5 && ActiveRecord::VERSION::MINOR >= 1 ||
-          (defined?(::ActiveRecord) && ActiveRecord::VERSION::MAJOR > 5)
-        "will_save_change_to_#{attr}?"
-      else
-        "#{attr}_changed?"
+    def algolia_attribute_changed?(object, attr_name)
+      # if one of two method is implemented, we return its result
+      # true/false means whether it has changed or not
+      ["#{attr_name}_changed?", "will_save_change_to_#{attr_name}?"].each do |method_name|
+        if object.respond_to?(method_name)
+          return object.send(method_name)
+        end
```

Comment:
Wouldn't this code essentially always call `#{attr_name}_changed?`, since on any random Rails model, calling `model.respond_to?(:xxx_changed?)` returns true for all model attributes?

If so, this would essentially nullify #325, which means that on Rails 5.1 this gem would revert to its old behavior i.e. it would start emitting warnings again.

As a reminder, the deprecation warning in Rails 5.1 was:

> DEPRECATION WARNING: The behavior of `attribute_changed?` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_change_to_attribute?` instead.

I believe this warning was removed in Rails 5.2 as they changed the behavior (I guess). But `xxx_changed?` still exists and will likely always exist.

Reply:
Yes, I agree. I don't know why I didn't see it myself. I guess after trying a few things, I started ignoring some cases. I was really looking to restore the previous behavior. 😅 
I definitely don't want to see those warning in more recent rails version 👍